src/node_buffer.cc

```diff
@@ -1211,6 +1211,41 @@ void DetachArrayBuffer(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void Btoa(const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 1);
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_IF_NOT_STRING(env, args[0], "argument");
+
+  Local<String> input = args[0].As<String>();
+  MaybeStackBuffer<char> buffer;
+  size_t written;
+
+  if (input->IsExternalOneByte()) {  // 8-bit case
+    auto ext = input->GetExternalOneByteStringResource();
+    size_t expected_length = simdutf::base64_length_from_binary(ext->length());
+    buffer.AllocateSufficientStorage(expected_length + 1);
+    buffer.SetLengthAndZeroTerminate(expected_length);
+    written = simdutf::binary_to_base64(
+        ext->data(), ext->length(), buffer.out(), simdutf::base64_default);
+  } else {  // 16-bit case
+    String::Value value(env->isolate(), input);
+    auto data = reinterpret_cast<const char*>(*value);
```

Comment:
This is definitely not the type of cast you want to make:

```js
$ node -p 'btoa("abcd")'
YWJjZA==
$ ./node -p 'btoa("abcd")'
YQBiAA==
```

Reply:
Thanks! Nice catch. Using Utf8Value fixes the problem. Would it be the best solution?

Reply:
Utf8Value is correct if and only if it's an ASCII-only string (if atob and btoa were actually using UTF-8, they wouldn't need to be deprecated)

The _best_ solution is probably to make use of `WriteOneByte()` instead of a `...Value` class